makeMLDoc.sh

```diff
@@ -1,57 +1,60 @@
+
+# GLM
+ecldoc --iroot ~/source/GLM --oroot ~/docs/GLM --eclcc=" -I ~/source" --exdocpaths docs/ML_Core/xml,docs/PBblas/xml --format "html,pdf" --exclude "performance/*.ecl,validation/*.ecl, IRLS/*.ecl,Datasets/*.ecl" --hideInternal
+
+# SupportVectorMachines
+ecldoc --iroot ~/source/SupportVectorMachines --oroot ~/docs/SupportVectorMachines --eclcc=" -I ~/source" --exdocpaths "docs/ML_Core/xml" --format "html,pdf" --exclude "Performance/*.ecl,Validation/*.ecl, Datasets/*.ecl, Examples/*.ecl, libsvm/*.ecl" --hideInternal
+
+# KMeans
+ecldoc --iroot ~/source/KMeans --oroot ~/docs/KMeans --eclcc=" -I ~/source" --exdocpaths docs/ML_Core/xml,docs/PBblas/xml --format "html,pdf" --exclude "test/*.ecl,test/datasets/*.ecl, test/Performance/*.ecl, test/Validation/*.ecl" --hideInternal
+
+# TextVectors
+ecldoc --iroot ~/source/TextVectors --oroot ~/docs/TextVectors --eclcc=" -I ~/source" --exdocpaths ~/docs/ML_Core/xml --format "html,pdf" --exclude "internal/*.ecl,test/*.ecl" --hideInternal
+
+# GNN
+ecldoc --iroot ~/source/GNN --oroot ~/docs/GNN --eclcc=" -I ~/source" --exdocpaths ~/docs/ML_Core/xml --format "html,pdf" --exclude "Internal/*.ecl,Test/*.ecl,Test/HARTests/*.ecl" --hideInternal
+
+# DBSCAN
+ecldoc --iroot ~/source/dbscan --oroot ~/docs/dbscan --eclcc=" -I ~/source" --exdocpaths ~/docs/ML_Core/xml --format "html,pdf" --exclude "internal/*.ecl,tests/*.ecl,tests/datasets/*.ecl" --hideInternal
+
+# HPCC-Causality
+ecldoc --iroot ~/source/HPCC_Causality --oroot ~/docs/HPCC_Causality --eclcc=" -I ~/source"  --format "html,pdf" --exclude "internal/*.ecl,Test/*.ecl,Test/Causality/*.ecl,Test/Probability/*.ecl,Test/Synth/*.ecl,performance/*.ecl,ecl/*.ecl" --hideInternal
+
+# GaussianProcessRegression
+ecldoc --iroot ~/source/GaussianProcessRegression --oroot ~/docs/GaussianProcessRegression --eclcc=" -I ~/source"  --format "html,pdf" --exclude "Internal/*.ecl,Test/*.ecl" --hideInternal
+
+# Now we copy the resulting index.pdf files to files in our publish area named correctly for each bundle.
+
+cp -f ~/docs/ML_Core/tex/index.pdf ~/share/Publish/ML_Core.pdf
+cp -f ~/docs/PBblas/tex/index.pdf ~/share/Publish/PBblas.pdf
+cp -f ~/docs/LinearRegression/tex/index.pdf ~/share/Publish/LinearRegression.pdf
+cp -f ~/docs/LogisticRegression/tex/index.pdf ~/share/Publish/LogisticRegression.pdf
+cp -f ~/docs/LearningTrees/tex/index.pdf ~/share/Publish/LearningTrees.pdf
+cp -f ~/docs/GLM/tex/index.pdf ~/share/Publish/GLM.pdf
+cp -f ~/docs/SupportVectorMachines/tex/index.pdf ~/share/Publish/SupportVectorMachines.pdf
+cp -f ~/docs/KMeans/tex/index.pdf ~/share/Publish/KMeans.pdf
+cp -f ~/docs/TextVectors/tex/index.pdf ~/share/Publish/TextVectors.pdf
+cp -f ~/docs/GNN/tex/index.pdf ~/share/Publish/GNN.pdf
+cp -f ~/docs/dbscan/tex/index.pdf ~/share/Publish/dbscan.pdf
+cp -f ~/docs/HPCC_Causality/tex/index.pdf ~/share/Publish/HPCC_Causality.pdf
+cp -f ~/docs/GaussianProcessRegression/tex/index.pdf ~/share/Publish/GaussianProcessRegression.pdf

```

Comment:
1. Add the change at the end instead of deleting everything and adding them back. 
2. Please update the documentation based on the latest GPR bundle update.

Reply:
@lilyclemson, after you updated the code 

I am hitting this error

```
enerating HTML Documentation ... 
Traceback (most recent call last):
  File "/usr/local/bin/ecldoc", line 9, in <module>
    main()
  File "/usr/local/bin/ecldoc", line 6, in main
    ecldoc.doMain()
  File "/usr/local/lib/python3.6/dist-packages/ecldoc/ecldoc.py", line 148, in doMain
    generators[f](input_root, output_root, ecl_file_tree, options).run()
  File "/usr/local/lib/python3.6/dist-packages/ecldoc/Formats/HTML/genHTML.py", line 216, in run
    self.gen('root', self.ecl_file_tree, self.html_root)
  File "/usr/local/lib/python3.6/dist-packages/ecldoc/Formats/HTML/genHTML.py", line 183, in gen
    license.text = '<a href="' + license.text + '">' + license.text + '</a>'
AttributeError: 'NoneType' object has no attribute 'text'
shivam@shivam-Virtual-Machine:~/source/GaussianProcessRegr
```

Reply:
Please find the possible cause and debug it. Thanks.